ecosystem/sf-indexer/aptos-sf-indexer/migrations/2022-08-08-043603_core_tables/up.sql

```diff
@@ -0,0 +1,285 @@
+  -- Constraints
+  PRIMARY KEY (key, sequence_number),
+  CONSTRAINT fk_transaction_versions FOREIGN KEY (transaction_version) REFERENCES transactions (version)
+);
+CREATE INDEX event_key_seq_type_index ON events (key, sequence_number, type);
+CREATE TABLE write_set_changes (
+  transaction_version BIGINT NOT NULL,
+  hash VARCHAR(255) NOT NULL,
+  transaction_block_height BIGINT NOT NULL,
+  type TEXT NOT NULL,
+  address VARCHAR(255) NOT NULL,
+  index BIGINT NOT NULL,
+  inserted_at TIMESTAMP NOT NULL DEFAULT NOW(),
+  -- Constraints
+  PRIMARY KEY (transaction_version, hash),
+  CONSTRAINT fk_transaction_versions FOREIGN KEY (transaction_version) REFERENCES transactions (version)
+);
+CREATE INDEX write_set_changes_tx_hash_addr_type_index ON write_set_changes (transaction_version, address, type);
+CREATE TABLE move_modules (
+  transaction_version BIGINT NOT NULL,
+  write_set_change_index BIGINT NOT NULL,
+  transaction_block_height BIGINT NOT NULL,
+  name VARCHAR(255) NOT NULL,
+  address VARCHAR(255) NOT NULL,
+  bytecode text,
+  friends jsonb,
+  exposed_functions jsonb,
+  structs jsonb,
+  is_deleted BOOLEAN NOT NULL,
+  inserted_at TIMESTAMP NOT NULL DEFAULT NOW(),
+  -- Constraints
+  PRIMARY KEY (transaction_version, write_set_change_index),
+  CONSTRAINT fk_transaction_versions FOREIGN KEY (transaction_version) REFERENCES transactions (version)
+);
+CREATE TABLE move_resources (
+  transaction_version BIGINT NOT NULL,
+  write_set_change_index BIGINT NOT NULL,
+  transaction_block_height BIGINT NOT NULL,
+  name VARCHAR(255) NOT NULL,
+  address VARCHAR(255) NOT NULL,
+  module VARCHAR(255) NOT NULL,
+  generic_type_params jsonb,
+  data jsonb,
+  is_deleted BOOLEAN NOT NULL,
+  inserted_at TIMESTAMP NOT NULL DEFAULT NOW(),
+  -- Constraints
+  PRIMARY KEY (transaction_version, write_set_change_index),
+  CONSTRAINT fk_transaction_versions FOREIGN KEY (transaction_version) REFERENCES transactions (version)
+);
+CREATE TABLE table_items (
+  key VARCHAR(255) NOT NULL,
+  transaction_version BIGINT NOT NULL,
+  write_set_change_index BIGINT NOT NULL,
+  transaction_block_height BIGINT NOT NULL,
+  table_handle VARCHAR(255) NOT NULL,
+  decoded_key jsonb NOT NULL,
+  decoded_value jsonb,
+  is_deleted BOOLEAN NOT NULL,
+  inserted_at TIMESTAMP NOT NULL DEFAULT NOW(),
+  -- Constraints
+  PRIMARY KEY (transaction_version, write_set_change_index),
+  CONSTRAINT fk_transaction_versions FOREIGN KEY (transaction_version) REFERENCES transactions (version)
+);
+CREATE TABLE table_metadatas (
+  handle VARCHAR(255) UNIQUE PRIMARY KEY NOT NULL,
+  key_type text NOT NULL,
+  value_type text NOT NULL,
+  inserted_at TIMESTAMP NOT NULL DEFAULT NOW()
+);
+CREATE TABLE indexer_states (
+  substream_module VARCHAR(100) NOT NULL,
+  block_height BIGINT NOT NULL,
+  success BOOLEAN NOT NULL,
+  details TEXT,
+  last_updated TIMESTAMP NOT NULL DEFAULT NOW(),
+  -- Constraints
+  PRIMARY KEY (substream_module, block_height)
```

Comment:
when debugging I'm always doing `where success is false`, would be great to have an index on it 😬